docs/04-reference/wingsdk-spec.md

```diff
@@ -253,105 +253,115 @@ resource Bucket {
   /**
    * Run an inflight whenever an object is uploaded to the bucket.
    */
-  onCreate(fn: inflight (key: str) => void, opts: BucketOnUploadProps?): void;
+  onCreate(fn: IBucketEventHandler, opts?: BucketOnUploadProps): void;
 
   /**
    * Run an inflight whenever an object is deleted from the bucket.
    */
-  onDelete(fn: inflight (key: str) => void, opts: BucketOnDeleteProps?): void;
+  onDelete(fn: IBucketEventHandler, opts?: BucketOnDeleteProps): void;
 
   /**
    * Run an inflight whenever an object is updated in the bucket.
    */
-  onUpdate(fn: inflight (key: str) => void, opts: BucketOnUpdateProps?): void;
+  onUpdate(fn: IBucketEventHandler, opts?: BucketOnUpdateProps): void;
 
   /**
    * Run an inflight whenever an object is uploaded, modified, or deleted from the bucket.
    */
-  onEvent(fn: inflight (event: BucketEvent) => void, opts: BucketOnEventProps?): void;
+  onEvent(fn: IBucketEventHandler, opts?: BucketOnEventProps?): void;
 
   /**
    * Add an object to the bucket that is uploaded when the app is deployed.
    */
-  addObject(key: str, value: Blob): void;
+  addObject(key: string, body: string): void;
```

Comment:
Since we don't have `Blob` type yet / adding it might require more design work, removing the uses of it makes sense to me 👍 
```suggestion
  addObject(key: str, body: str): void;
```